parsl/executors/status_handling.py

```diff
@@ -87,6 +87,17 @@ def tasks(self) -> Dict[object, Future]:
     def provider(self):
         return self._provider
 
+    def _filter_scale_in_ids(self, to_kill, killed):
+        assert len(to_kill) == len(killed)
+
+        result = []
+
+        for i in range(len(killed)):
```

Comment:
If this is filtering, it might be clearer to use itertools.compress : 
```
from itertools import compress
result = list(compress(to_kill, killed)) # Filters first iterable by bool values in second.
```

Reply:
Sure. I started that way, but then I didn't think "compress" was standard FP naming (Ben can correct me if I'm wrong), and thought that being clear is better than being compact. But it looks like a carefully placed comment may do the trick.

Reply:
To be clear, by "that way" I meant

 `reduce(lambda l, x: l + [x[0]] if x[1] else l, zip(to_kill, killed), [])`

This is clearly not the same as using `compress`, whose name also doesn't have to match what I thought was akin to a `filter` function, but isn't.

Reply:
I haven't seen `compress` in other FP situations.

I don't have a lot of opinion either way, but I think I slightly prefer the compress style...

Reply:
@hategan Can we do compress with a comment to clarify?